atlp-rwanda/tech-strikers-bn-backend
# #17563684 Login User and Generate a Token When Password is Excluded

#### What does this PR do?
Login user to the application, and generate a token containing information with password excluded
#### Description of Task to be completed?

-  check the user existence in the database via the registered email address
-  end the process by sending error if the user is not in the database
-  validate required credentials 
-  login the user and generate the token containing information with password excluded in the token
- write tests to test login functionality

#### How should this be manually tested?

- run `git clone https://github.com/atlp-rwanda/tech-strikers-bn-backend.git`
- run `cd tech-strikers-bn-backend`
- run `git checkout ft-userAuthentication-#17563684`
- run `npm install`
- create  your .env file from .env.example file and add variable ACCESS_TOKEN_SECRET in that file with your random value
- add database username, password and database name in your .env file
- run `sequelize db:migrate`
- run `sequelize db:seed:all` in order to have the seeds and apply login on postman
- run `npm run dev`
- use postiman with `localhost:3000/api/auth/siginIn` to login the existing user in the migrated database by using the following credentials: 
- use `{   
  "email": "dusaflora2@gmail.com",
   "password": "tytyne12345"
}`
- run `npm run test` to test the login API. It should pass when email and password are valid
#### Any background context you want to provide?
Even though my login API test is passing, there are other uncovered lines that need to be tested in the coverage table.
#### What are the relevant pivotal tracker stories?
[#17563684](https://www.pivotaltracker.com/n/projects/2475704/stories/175636843)
#### Screenshots (if appropriate)
<img width="1150" alt="login user" src="https://user-images.githubusercontent.com/56275083/102866219-90ecd280-443f-11eb-9f1a-56cbe6a77bac.png">
